### PR TITLE
results_table

### PR DESCRIPTION
**Pull request by daler**
This is a cleaned-up version of the `metaseq.results_table` module for flexible plotting with marginal histograms and callback functions (by default, print the pandas.DataFrame row of the point that was clicked).

Still needs a lot of testing before merging. In the past I've tested this module sort of by hand and by using it in everyday work. Not sure the best way to test it now, so that'll take some playing around.

### CODE REVIEW

**Comment by daler** [comment]
Tests added. As we discussed it's tough to do a good job testing plots. For now I'm must making sure the plotting code runs rather than test for correctness with e.g. PNG comparisons, but the non-plotting code is mostly tested.

File: lcdblib/plotting/results_table.py [hunk 1]
```diff
@@ -0,0 +1,1265 @@
...
+
+    def __init__(self, data, db=None, header_check=True, **kwargs):
+        import_kwargs = kwargs.pop('import_kwargs', {})
+        if header_check and isinstance(data, str):
+            comment_char = import_kwargs.get('comment', '#')
+            for i, line in enumerate(open(data)):
+                if line[0] != comment_char:
+                    break
+            import_kwargs['skiprows'] = i
+        import_kwargs['na_values'] = ['nan']
```
Comment (jfear) [inline lcdblib/plotting/results_table.py]: Doesn't pandas.read_table() given a comment character do the same thing?

File: lcdblib/plotting/results_table.py [hunk 2]
```diff
@@ -0,0 +1,1265 @@
...
+        """
+        Generate the autosql for DESeq results (to create bigBed)
+
+        Returns a temp filename containing the autosql defining the extra
+        fields.
+
+        This for creating bigBed files from BED files created by
+        colormapped_bed.  When a user clicks on a feature, the DESeq results
+        will be reported.
+        """
```
Comment (jfear) [inline lcdblib/plotting/results_table.py]: This is rather slick

File: lcdblib/plotting/results_table.py [hunk 3]
```diff
@@ -0,0 +1,1265 @@
...
+            ensure all ResultsTable objects are aligned to the same index. If
+            you don't want to provide this, then set it to None and consider
+            the `index_from` kwarg.
+
+        index_from : str
+            Key of the dataframe whose index will be used to align other
+            dataframes. If this is provided, this dataframe will have to be
+            loaded before any others, so the first access will trigger two
+            dataframe loads.
+
```
Comment (jfear) [inline lcdblib/plotting/results_table.py]: You don't have the index_from kwarg in the init.

File: lcdblib/plotting/results_table.py [hunk 4]
```diff
@@ -0,0 +1,1265 @@
...
+        ax.figure.canvas.mpl_connect('pick_event', wrapped_callback)
+
+        ax.set_xlabel(xlab)
+        ax.set_ylabel(ylab)
+        ax.axis('tight')
+
+        return ax
+
+    def radviz(self, column_names, transforms=dict(), **kwargs):
+        """
```
Comment (daler) [inline lcdblib/plotting/results_table.py]: The pandas function does a lot of stuff on its own, and I wanted to be able to do things like attach callbacks (for clicking on points in the callback) and to provide a similar API as to `ResultsTable.scatter`.

File: lcdblib/plotting/results_table.py (hunk 1 above)
Comment (daler) [inline lcdblib/plotting/results_table.py]: Yep. This was copied from metaseq. My guess is that the comment char wasn't added to pandas back then? I'll fix it.

File: lcdblib/plotting/results_table.py (hunk 2 above)
Comment (daler) [inline lcdblib/plotting/results_table.py]: Thanks! Still has issues with corner cases (all up, or all down, or nothing changed) and should expose a Normalize instance config to be able to make colormaps the same across multiple experiments. But it's a nice way of zooming to interesting places in the browser without needing collaborators to dig through the tsv of results.

File: lcdblib/plotting/results_table.py (hunk 3 above)
Comment (daler) [inline lcdblib/plotting/results_table.py]: Oops thanks. This definitely needs more tests.

**Pull request merged**